pmlc/conversion/tile_to_pxa/tile_to_pxa.cc

```diff
@@ -1362,35 +1364,111 @@ struct ScatterOpConversion : public OpConversionPattern<tile::ScatterOp> {
 
     // Load the location value from the indices tensor.
     // Create an affine map for loading the index, using leading counters.
-    size_t idxDims = indices.getType().cast<MemRefType>().getShape().size();
+    int axis = *(op.axis().getRawData());
+    size_t dstDims = resultMemRefType.getShape().size();
+    // TODO: Deal with negative axis earlier in ast?
+    if (axis < 0) {
+      axis = dstDims + axis;
+    }
+
+    auto idxShape = indices.getType().cast<MemRefType>().getShape();
+    size_t idxDims = idxShape.size();
     auto idxLoadMap = AffineMap::getMultiDimIdentityMap(idxDims, ctx);
-    auto idxLoadOps = loop.getIVs().take_front(idxDims);
+    SmallVector<Value, 4> dstOps;
+    // TODO: Move this code block into separate functions?
```

Comment:
yes please move the if else block to a new function, and use 
`if(op.updateMode() == ScatterUpdateMode::nd)) {}else{}`
instead of 
`if(op.updateMode() != ScatterUpdateMode::nd)) {}else{}`